proposals/NNNN-remove-policy-mode-option-from-sdl-core.md

```diff
@@ -0,0 +1,286 @@
+
+Generally speaking, `EXTERNAL_PROPRIETARY` mode is the fully-featured version of policies, while the other two modes only support a subset of the features in `EXTERNAL_PROPRIETARY`. There is a significant amount of overhead in testing and development for policy-related features due to the need to support each of these modes. For these reasons, there is very little reason to continue supporting these older modes, as they are less versatile than `EXTERNAL_PROPRIETARY` mode. The goal of this proposal is to remove the `EXTENDED_POLICY` build option from SDL Core, reducing support only to `EXTERNAL_PROPRIETARY` policies.
+
+## Motivation
+
+Currently, the codebase for policies in SDL Core is split between two folders with similar contents:
+
+- `policy/policy_regular` contains the codebase for both `HTTP` and `PROPRIETARY` mode.
+- `policy/policy_external` contains the codebase for `EXTERNAL_PROPRIETARY` mode.
+
+This means that each time an addition to the policy table is made, the same changes need to be made in both places, significantly increasing development time. 
+
+In addition, having three possible configurations for the policy mode significantly increases the amount of time needed to test policy-related features. Each policy feature must be tested in all three modes, requiring three separate full builds for the feature.
+
+Overall, there seems to be very little benefit to supporting anything besides `EXTERNAL_PROPRIETARY` mode, since it is more versatile and feature-rich than either of the other modes, while the costs of maintaining all three modes is fairly steep.
+
+## Proposed solution
+
+The proposed solution to this problem is to remove the `EXTENDED_POLICY` build flag from the SDL Core project entirely, keeping only the implementation for `EXTERNAL_PROPRIETARY` mode. This would reduce the complexity of the policy component of SDL Core quite significantly, speeding up development and testing for this component while reducing the chance of issues caused by any differences in the implementation of each policy mode.
+
+## Potential downsides
+
+If any existing systems use either `HTTP` or `PROPRIETARY` policy mode, they will need to modify their HMI to work with the `EXTERNAL_PROPRIETARY` policy update flow. This would not require a significant HMI overhaul, but is still something to keep in mind. Another note is that unfortunately `PROPRIETARY` is the default policy mode at the moment, meaning that it is quite possible that this mode is used by existing systems. One thing that could be done to help with this downside would be to include a HMI guide for transitioning to the `EXTERNAL_PROPRIETARY` policy flow.
```

Comment:
```suggestion
If any existing systems use either `HTTP` or `PROPRIETARY` policy mode, they will need to modify their HMI to work with the `EXTERNAL_PROPRIETARY` policy update flow. This would not require a significant HMI overhaul, but is still something to keep in mind. Another note is that unfortunately `PROPRIETARY` is the default policy mode at the moment, meaning that it is quite possible that this mode is used by existing systems. One thing that could be done to help with this downside would be to include an HMI guide for transitioning to the `EXTERNAL_PROPRIETARY` policy flow.
```

Reply:
Adressed in https://github.com/smartdevicelink/sdl_evolution/pull/1170/commits/595aeb015144893eb5748a656cdf28793d49856a